workflow/rules/authentic.smk

```diff
@@ -115,67 +114,62 @@ rule Post_Processing:
 
 rule Samtools_Faidx:
     output:
-        fai="{prefix}.{fasta}.fai"
+        fai="{prefix}.{fasta}.fai",
     input:
-        fna="{prefix}.{fasta}"
+        fna="{prefix}.{fasta}",
     wildcard_constraints:
-        fasta="(fna|fasta|fa)"
+        fasta="(fna|fasta|fa)",
     message:
         "Samtools_Faidx: INDEXING MALT FASTA DATABASE FOR BREADTH_OF_COVERAGE SEQUENCE RETRIEVAL"
     log:
-        "logs/BREADTH_OF_COVERAGE/{prefix}.{fasta}.fai.log"
+        "logs/BREADTH_OF_COVERAGE/{prefix}.{fasta}.fai.log",
     conda:
         "../envs/samtools.yaml"
     envmodules:
-        *config["envmodules"]["samtools"]
+        *config["envmodules"]["samtools"],
     shell:
         "samtools faidx {input.fna}"
 
 
 rule Breadth_Of_Coverage:
     input:
-        nodeentries="results/AUTHENTICATION/{sample}/{taxid}/{sample}.trimmed.rma6_MaltExtract_output/default/readDist/{sample}.trimmed.rma6_additionalNodeEntries.txt",
         sam="results/MALT/{sample}.trimmed.sam.gz",
         malt_fasta=config["malt_nt_fasta"],
-        malt_fasta_fai=f"{config['malt_nt_fasta']}.fai"
+        malt_fasta_fai=f"{config['malt_nt_fasta']}.fai",
     output:
-        name_list="results/AUTHENTICATION/{sample}/{taxid}/{refid}/name.list",
-        sam="results/AUTHENTICATION/{sample}/{taxid}/{refid}/{taxid}.sam",
-        bam="results/AUTHENTICATION/{sample}/{taxid}/{refid}/{taxid}.bam",
-        sorted_bam="results/AUTHENTICATION/{sample}/{taxid}/{refid}/{taxid}.sorted.bam",
-        breadth_of_coverage="results/AUTHENTICATION/{sample}/{taxid}/{refid}/{taxid}.breadth_of_coverage",
-        fasta="results/AUTHENTICATION/{sample}/{taxid}/{refid}/{taxid}.fasta",
+        name_list="results/AUTHENTICATION/{sample}/{taxid}/name_list.txt",
+        sorted_bam="results/AUTHENTICATION/{sample}/{taxid}/sorted.bam",
+        breadth_of_coverage="results/AUTHENTICATION/{sample}/{taxid}/breadth_of_coverage",
     params:
         ref_id=get_ref_id,
     message:
         "Breadth_Of_Coverage: COMPUTING BREADTH OF COVERAGE, EXTRACTING REFERENCE SEQUENCE FOR VISUALIZING ALIGNMENTS WITH IGV"
     log:
-        "logs/BREADTH_OF_COVERAGE/{sample}_{taxid}_{refid}.log",
+        "logs/BREADTH_OF_COVERAGE/{sample}_{taxid}.log",
     conda:
         "../envs/malt.yaml"
     envmodules:
         *config["envmodules"]["malt"],
     shell:
         "echo {params.ref_id} > {output.name_list}; "
-        "zgrep {params.ref_id} {input.sam} | uniq > {output.sam}; "
-        "samtools view -bS {output.sam} > {output.bam}; "
-        "samtools sort {output.bam} > {output.sorted_bam}; "
+        "zgrep {params.ref_id} {input.sam} | uniq > results/AUTHENTICATION/{wildcards.sample}/{wildcards.taxid}/{params.ref_id}.sam; "
+        "samtools view -bS results/AUTHENTICATION/{wildcards.sample}/{wildcards.taxid}/{params.ref_id}.sam > results/AUTHENTICATION/{wildcards.sample}/{wildcards.taxid}/{params.ref_id}.bam; "
```

Comment:
Also, I think in the future that we could split up some of the long multi-command rules to separate rules, where intermittent file removal could be handled automatically.

Reply:
The sam file was left in the directory even before, if I understand correctly what you mean. Is this something that should be removed instead?

Reply:
Yes, that's what I meant. Since we have the bam file, do we even need to keep it around? It is not the most space-efficient file format. This is where it could come in handy having a separate rule to generate sam, with an output that is temporary. I'll add an issue for reference.

Reply:
Ok, might be good to see what @LeandroRitter thinks here

Reply:
One could also here add the sam file as a temporary output of Breadth_Of_Coverage.

Reply:
Do we mark as temporary only `refid.sam` or both that and `sorted.sam`?

Reply:
If we don't need them downstream (i.e. there are bam files for both) then I'd say mark both as temporary. If they are needed downstream, be it for viewing alignments, they should at the very least be explicitly listed as output files, IMO.

Reply:
This can be done if I change the filename back to something like {taxid}.sam rather than {refid}.sam (we can't use refid in the output section). I think the new commit solves it